### PR TITLE
Metadata visualization improvement

### LINKED ISSUE #54: Next steps for improvements of all demo progress bars
Very relevant to how this should ultimately appear on the GUIDE...

Can you add 
- text to the left of each bar drawing from the `desc` of the `format_dict`?
- text counter of `n` and `total` as `f"{n}/{total} ({n/total:.2f}%)"` - can either be inside the bar or to the right of the bar, or whoever else you think would be visually appealing
- estimate of speed per update and time remaining (`remaining` and `rate_fmt` of the `format_dict` should be helpful for this)

Basically trying to map as much as possible between a classic console TQDM and the much prettier front-end one

### PR DESCRIPTION
**Pull request by garrettmflynn**
fix #54

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
OOooo, something else I forgot to requestt - would it be possible to add a 'cancel' button on the frontend instead of having to ctrl+C the server each time? (could be useful in the GUIDE as well to cancel a conversion/inspection/upload operation to the best of our ability

**Comment by CodyCBakerPhD** [comment]
@garrettmflynn Reminder on conflicts here - let me know when you removed the websocket stuff too and I'll take a look at this

**Comment by garrettmflynn** [comment]
Good to go! Haven't gotten around to the cancel request yet—but I'll see what I can do.

**Comment by garrettmflynn** [comment]
> OOooo, something else I forgot to requestt - would it be possible to add a 'cancel' button on the frontend instead of having to ctrl+C the server each time? (could be useful in the GUIDE as well to cancel a conversion/inspection/upload operation to the best of our ability

Hmm this is somewhat complicated given that HTTP requests are one-off. I've seen a few [hacks](https://stackoverflow.com/questions/18511119/stop-processing-flask-route-if-request-aborted) online that indicate that you can detect if the client aborts the request while it's occuring—but these are not super clean.

The only other way to do this would be to maintain global references to the running bars / processes and have a `/cancel` endpoint that can clean these up. 

Are these solutions something that you would be interested in?

**Comment by CodyCBakerPhD** [comment]
There's no way to track the PIDs of the processes and send a `pkill` or similar signal? I know how that might be done on the Python side (so through an endpoint) if JS doesn't have a way - in that case you'd have a 'cancel' button that would call the endpoints which would trigger the process kill?

**Comment by garrettmflynn** [comment]
Using that approach, we could notify the browser of related `pids` using SSE events as the processes are created. Though AFAIK this wouldn't abort the endpoint execution itself.

Would that actually achieve what you want? And I'm assuming you really only care about canceling in the parallel demo?

Otherwise, we could _probably_ execute everything in this endpoint as a thread, then use the hacks referenced before to send some kill command to that—which would have references to everything directly and could shut it down that way.

**Comment by CodyCBakerPhD** [comment]
OK after digging deeper into cancellation, the only/best 'safe' way to do this would be to have it as a specific feature of the iteration process on the backend

I have an idea for how to implement this in a follow up so I'll kick the bucket down the chain for now

File: src/tqdm_publisher/_demos/_parallel_bars/_server.py [hunk 1]
```diff
@@ -127,13 +125,13 @@ def run_parallel_processes(*, all_task_times: List[List[float]], request_id: str
 
         total_tasks_iterable = as_completed(futures)
         total_tasks_progress_bar = TQDMProgressPublisher(
-            iterable=total_tasks_iterable, total=len(all_task_times), desc="Total tasks completed"
+            iterable=total_tasks_iterable, total=len(all_task_times), desc=f"Total tasks completed for {request_id}"
```
Comment (CodyCBakerPhD) [inline src/tqdm_publisher/_demos/_parallel_bars/_server.py:128]: @garrettmflynn The frontend display of the subprogress bars looks great!

Can you also add the same UI for the 'main' (top) progress bar too? One test is that this demo should use the custom `desc` seen here

Comment (garrettmflynn) [reply src/tqdm_publisher/_demos/_parallel_bars/_server.py:128]: Technically this was already happening, though the main description has been moved up beneath the header. Do you want it moved back? 

Screenshot 2024-04-13 at 8 58 52 AM

Additionally, all bar information is only popupated on the _first_ update—and the main progress bar takes a while (i.e. until one of the sub-bars finishes).

I've updated the `TQDMProgressPublisher` to send the current state immediately after subscribing? Is that alright or do we want to handle that manually in each demo?

Comment (CodyCBakerPhD) [reply src/tqdm_publisher/_demos/_parallel_bars/_server.py:128]: Definitely better - could you move the `desc` and elapsed/remaining to under the main bar to stylistically match the others?

**Comment by garrettmflynn** [comment]
Done!

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**